File: src/easy/0225-implement-stack-using-queues.cpp
class MyStack {
public:
	/** Initialize your data structure here. */
	int queue[1000];
	int rear;
	int front;

	MyStack() {
		rear = front = -1;
	}

	/** Push element x onto stack. */
	void push(int x) {
		queue[++rear] = x;
	}

	/** Removes the element on top of the stack and returns that element. */
	int pop() {
		return queue[rear--];
	}

	/** Get the top element. */
	int top() {
		return queue[rear];
	}

	/** Returns whether the stack is empty. */
	bool empty() {
		return (rear == front);
	}
};

File: src/easy/0225-implement-stack-using-queues.js
/**
 * Initialize your data structure here.
 */
var MyStack = function() {
  this.data = [];
};

/**
 * Push element x onto stack. 
 * @param {number} x
 * @return {void}
 */
MyStack.prototype.push = function(x) {
  this.data.push(x);
};

/**
 * Removes the element on top of the stack and returns that element.
 * @return {number}
 */
MyStack.prototype.pop = function() {
  return this.data.pop();
};

/**
 * Get the top element.
 * @return {number}
 */
MyStack.prototype.top = function() {
  return this.data[this.data.length - 1];
};

/**
 * Returns whether the stack is empty.
 * @return {boolean}
 */
MyStack.prototype.empty = function() {
  return this.data.length === 0;
};

/** 
 * Your MyStack object will be instantiated and called as such:
 * var obj = Object.create(MyStack).createNew()
 * obj.push(x)
 * var param_2 = obj.pop()
 * var param_3 = obj.top()
 * var param_4 = obj.empty()
 */